src/output/output_precedence.rs

```diff
@@ -0,0 +1,95 @@
+//! Defines the precedence of the output keys in the session outputs.
+//!
+//! # Note
+//!
+//! The purpose of this module is to replicate the [existing output key selection mechanism]
+//! in the library. This is an acceptable solution in lieu of a model-specific solution,
+//! e.g. reading the output keys from the model file.
+//!
+//! [existing output key selection mechanism]: OutputPredecence::DEFAULT
+
+use ndarray::{Array, Dim, IxDynImpl};
+
+/// Enum for defining the key of the output.
+#[derive(Debug, Clone)]
+pub enum OutputKey {
+    OnlyOne,
+    ByOrder(usize),
+    ByName(&'static str),
+}
+
+impl Default for OutputKey {
+    fn default() -> Self {
+        Self::OnlyOne
+    }
+}
+
+/// Trait for defining a precedence of keys in the output.
+///
+/// This defines the order of precedence for selecting the output from the session outputs.
+/// By convention, an ONNX model will have at least one output called `last_hidden_state`,
+/// which is however not guaranteed. This trait allows the user to define the order of
+/// precedence for selecting the output.
+///
+/// Any [`OutputPrecedence`] should be usable multiple times, and should not consume itself;
+/// this is due to use of [`rayon`] parallelism, which means
+/// [`OutputPrecedence::key_precedence`] will have to be called once per batch.
+pub trait OutputPrecedence {
+    /// Get the precedence of the keys in the output.
+    fn key_precedence(&self) -> impl Iterator<Item = &OutputKey>;
+
+    /// Default precedence of the output keys.
+    ///
+    /// This will select the output based on the following precedence:
+    /// 1. If there is only one output, select that output.
+    /// 2. If there are multiple outputs, select the output by the name `last_hidden_state`.
+    /// 3. If the above two conditions are not met, return an error.
+    ///
+    /// This is the current precedence used by the library.
+    const DEFAULT: &'static [OutputKey] =
+        &[OutputKey::OnlyOne, OutputKey::ByName("last_hidden_state")];
```

Comment:
Unused?

Reply:
I left that in almost as a documentation of existing behaviour, and potentially use it in `sparse_text_embedding`. Mostly because I can't quite understand how these keys work, or why `last_hidden_state` is a chosen special key. I was trying to find documentations but ended up rather more confused than enlightened...

So I imagine a user will be equally baffled by this, and giving them a `DEFAULT` or at least and an example of what to expect will help, hence the detailed explanation of what its doing.

Reply:
I think https://github.com/denwong47/fastembed-rs/blob/e0f338508d6fb515c6a3c6c0cc469ae91a44315f/src/text_embedding.rs#L97-L104 is pretty explanatory and will cover for this.

Reply:
Sure, I can remove it.